src/SprykerEco/Zed/PunchoutCatalogs/Business/Reader/PunchoutCatalogsReaderInterface.php

```diff
@@ -24,4 +26,20 @@ public function findConnectionById(int $idConnection): ?PunchoutCatalogConnectio
      * @return \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer|null
      */
     public function findConnectionByIdWithPassword(int $idConnection): ?PunchoutCatalogConnectionTransfer;
+
+    /**
+     * @param int $fkCompanyBusinessUnit
+     *
+     * @return \Generated\Shared\Transfer\PunchoutCatalogConnectionCollectionTransfer|null
+     */
+    public function findConnectionByFkCompanyBusinessUnit(
```

Comment:
incorrect approach used
you either need to `findConnectionByIdCompanyBusinessUnit(): ?PunchoutCatalogConnectionTransfer `
or `getPunchoutCatalogConnectionsByIdCompanyBusinessUnit: PunchoutCatalogConnectionCollectionTransfer `

Reply:
Fell free to ask questions

Reply:
Reworked.